mods/default/nodes.lua

```diff
@@ -1534,13 +1512,56 @@ minetest.register_node("default:torch", {
 	light_source = default.LIGHT_MAX - 1,
 	selection_box = {
 		type = "wallmounted",
-		wall_top = {-0.1, 0.5 - 0.6, -0.1, 0.1, 0.5, 0.1},
-		wall_bottom = {-0.1, -0.5, -0.1, 0.1, -0.5 + 0.6, 0.1},
-		wall_side = {-0.5, -0.3, -0.1, -0.5 + 0.3, 0.3, 0.1},
+		wall_top = {-1/16, -2/16, -1/16, 1/16, 0.5, 1/16},
+		wall_bottom = {-1/16, -0.5, -1/16, 1/16, 2/16, 1/16},
 	},
 	groups = {choppy = 2, dig_immediate = 3, flammable = 1, attached_node = 1},
-	legacy_wallmounted = true,
 	sounds = default.node_sound_defaults(),
+	drop = "default:torch",
+	on_place = function(itemstack, placer, pointed_thing)
+		local above, under = pointed_thing.above, pointed_thing.under
```

Comment:
`minetest.registered_nodes[name]`

Provides you a nodedef, but it can be `nil` if you're looking up an unknown node.

You could also use `minetest.get_node_or_nil()` here to get the name, which will return `nil`, if you right click an unloaded node. While not strictly necessary here, due to lack of node setting; it's still a bit cleaner imo.